python/ray/train/torch/torch_trainer.py

```diff
@@ -16,217 +16,121 @@
 
             import ray
-            from ray import train
             from ray.train import Checkpoint, CheckpointConfig, RunConfig, ScalingConfig
             from ray.train.torch import TorchTrainer
 
             # If using GPUs, set this to True.
-            use_gpu = False
-
-            # Define NN layers archicture, epochs, and number of workers
-            input_size = 1
-            layer_size = 32
-            output_size = 1
-            num_epochs = 20
-            num_workers = 3
+            use_gpu = True
+            num_workers = 4
 
             # Define your network structure
             class NeuralNetwork(nn.Module):
                 def __init__(self):
                     super(NeuralNetwork, self).__init__()
-                    self.layer1 = nn.Linear(input_size, layer_size)
+                    self.layer1 = nn.Linear(1, 32)
                     self.relu = nn.ReLU()
-                    self.layer2 = nn.Linear(layer_size, output_size)
+                    self.layer2 = nn.Linear(32, 1)
 
                 def forward(self, input):
                     return self.layer2(self.relu(self.layer1(input)))
 
-            # Define your train worker loop
-            def train_loop_per_worker():
-                torch.manual_seed(42)
+            # Training loop.
+            def train_loop_per_worker(config):
+
+                # Read configurations.
+                lr = config["lr"]
+                batch_size = config["batch_size"]
+                num_epochs = config["num_epochs"]
 
-                # Fetch training set
-                dataset_shard = train.get_dataset_shard("train")
+                # Fetch training dataset.
+                train_dataset_shard = ray.train.get_dataset_shard("train")
+
+                # Setup model.
                 model = NeuralNetwork()
+                model = ray.train.torch.prepare_model(model)
 
-                # Loss function, optimizer, prepare model for training.
-                # This moves the data and prepares model for distributed
-                # execution
+                # Define loss and optimizer.
                 loss_fn = nn.MSELoss()
-                optimizer = torch.optim.Adam(model.parameters(),
-                            lr=0.01,
-                            weight_decay=0.01)
-                model = train.torch.prepare_model(model)
-
-                # Iterate over epochs and batches
-                for epoch in range(num_epochs):
-                    for batches in dataset_shard.iter_torch_batches(batch_size=32,
-                                dtypes=torch.float):
+                optimizer = torch.optim.SGD(model.parameters(), lr=lr)
 
-                        # Add batch or unsqueeze as an additional dimension [32, x]
-                        inputs, labels = torch.unsqueeze(batches["x"], 1), batches["y"]
-                        output = model(inputs)
+                # Create data loader.
```

Comment:
Does everyone know what a data loader is?

Reply:
Yeah this is standard PyTorch terminology.